opensfm/io.py

```diff
@@ -57,27 +58,39 @@ def camera_from_json(key, obj):
     return camera
 
 
-def shot_from_json(key, obj, cameras):
+def shot_from_json(reconstruction, key, obj, is_pano_shot=False):
     """
     Read shot from a json object
     """
-    pose = types.Pose()
+    pose = pygeometry.Pose()
     pose.rotation = obj["rotation"]
     if "translation" in obj:
         pose.translation = obj["translation"]
 
-    metadata = types.ShotMetadata()
-    metadata.orientation = obj.get("orientation")
-    metadata.capture_time = obj.get("capture_time")
-    metadata.gps_dop = obj.get("gps_dop")
-    metadata.gps_position = obj.get("gps_position")
-    metadata.skey = obj.get("skey")
-
-    shot = types.Shot()
-    shot.id = key
+    metadata = pymap.ShotMeasurements()
+    if obj.get("orientation") is not None:
+        metadata.orientation.value = obj.get("orientation")
+    if obj.get("capture_time") is not None:
```

Comment:
I wonder if there would be a way to translate  some C++14 `Optional<T>` to corresponding `python` type or `None` :scratching_head_smiley: